### DISCUSSION TITLE
Comfy UI :  ModuleNotFoundError: No module named 'safetensors'

### DISCUSSION THREAD

**Discussion by Autantpourmoi** [Comfy-Org/ComfyUI#1427] [General]
After having some pb with homebrew after migrating my MBP intel to MBP M2 , i had to uninstall Comfy UI ( who was running well before th pb) , i reinstall it but can't make it run , i always had this error message : no module named 'safetensors'
if someone can give me some help that will be greatly appreciated

python3 main.py
** ComfyUI start up time: 2023-09-05 19:38:58.561372

Prestartup times for custom nodes:
   0.0 seconds: /Users/lucvaligny/ComfyUI/custom_nodes/ComfyUI-Manager

Traceback (most recent call last):
  File "/Users/lucvaligny/ComfyUI/main.py", line 66, in <module>
    import comfy.utils
  File "/Users/lucvaligny/ComfyUI/comfy/utils.py", line 5, in <module>
    import safetensors.torch
ModuleNotFoundError: No module named 'safetensors'

**Comment by majunchi0000** [comment]
Guys,I also encountered this problem and wanted to ask if it was solved?

**Comment by ltdrdata** [comment]
You didn't installed ComfyUI properly.
https://github.com/comfyanonymous/ComfyUI#manual-install-windows-linux

If you can set up a `venv`, you need to install the `requirements.txt` and other packages using pip. If you are not familiar with setting up a venv, it is recommended to just use the portable version.

**Comment by Autantpourmoi** [comment]
like you guys , i wasn't understanding anything with this venv virtual environment these 2 links helps me a lot 
https://medium.com/artificialis/virtualenv-and-local-environment-in-python-and-pip-424c68d8bc2d
https://sourabhbajaj.com/mac-setup/Python/virtualenv.html
i was able to create a venv after get clone Comfy ui after reading those 2 pages and make it work for my Mac OS installation

**Comment by sguzman** [comment]
I don't think it's a matter of not following the directions. I followed them explicitly as posted on the readme in the "Installing" for "Manual Install (Windows, Linux)" and still got the same error. I am using Linux (arch) and using micromamba to manage my environment. I created a specific environment for comfyui, ran `pip install torch torchvision torchaudio --extra-index-url https://download.pytorch.org/whl/cu121` since I use a NVIDIA gpu and installed the `requirements.txt` dependencies. I tried python-3.11 and python-3.10.

Still, I got the same error. Currently, I am trying to troubleshoot myself. I will update if I manage to fix it for my environment.

**Comment by mitong2023** [comment]
Python：
pip install safetensors

if have different version
pip3 install safetensors
python -m pip install safetensors
python3 -m pip install safetensors

 Linux 
sudo pip3 install safetensors
pip3 install safetensors --user

  如果你有 Linux 和 apt
sudo apt install safetensors

Windows and py have alias
py -m pip install safetensors

 Anaconda
conda install -c anaconda safetensors

  Jupyter Notebook
!pip install safetensors
!pip3 install safetensors
